src/ESM.sol

```diff
@@ -56,22 +62,27 @@ contract ESM {
     }
 
     function fire() external {
-        require(fired == 0,  "esm/already-fired");
-        require(Sum >= min,  "esm/min-not-reached");
+        require(Sum >= min,  "ESM/min-not-reached");
 
         end.cage();
+        if (proxy != address(0)) {
+            VatLike(end.vat()).deny(proxy);
+        }
 
-        fired = 1;
         emit Fire();
     }
 
     function join(uint256 wad) external {
-        require(fired == 0, "esm/already-fired");
+        require(end.live() == 1, "ESM/system-already-shutdown");
 
         sum[msg.sender] = add(sum[msg.sender], wad);
         Sum = add(Sum, wad);
 
-        require(gem.transferFrom(msg.sender, pit, wad), "esm/transfer-failed");
+        require(gem.transferFrom(msg.sender, address(this), wad), "ESM/transfer-failed");
         emit Join(msg.sender, wad);
     }
+
+    function burn() external {
```

Comment:
We might want to tell Oasis about this option, but it shouldn't stop the function of `join()` and `burn()`.